### ISSUE TITLE
Guitar notation with lilypond

### ISSUE THREAD

**Issue by casertap** [patacrep/patacrep#221]
I was using the songbook many years ago and decided to get back to my instrument :-)
I think you did a great job refactoring the software. Awesome job, thanks a lot.
I have trouble importing my old songs to the news format, my lilypond looks funny.

before I had lilypond that looked like this: 
````
#(define (tie::tab-clear-tied-fret-numbers grob)
   (let* ((tied-fret-nr (ly:spanner-bound grob RIGHT)))
      (ly:grob-set-property! tied-fret-nr 'transparent #t)))

\version "2.13.22" 
\include "header"
\paper{
paper-width = 8\cm
line-width = 8\cm
paper-height=5\cm
}
\layout {
   \context { \Score
      \override MetronomeMark #'padding = #'5
   }
   \context { \Staff
      \override TimeSignature #'style = #'numbered
      \override StringNumber #'transparent = ##t
   }
   \context { \TabStaff
      \override TimeSignature #'style = #'numbered
      \override Stem #'transparent = ##f
      \override Beam #'transparent = ##f
      \override Tie  #'after-line-breaking = #tie::tab-clear-tied-fret-numbers
   }
   \context { \StaffGroup
      \consists "Instrument_name_engraver"
   }
}
deadNote = #(define-music-function (parser location note) (ly:music?)
   (set! (ly:music-property note 'tweaks)
      (acons 'stencil ly:note-head::print
         (acons 'glyph-name "2cross"
            (acons 'style 'special
               (ly:music-property note 'tweaks)))))
   note)

palmMute = #(define-music-function (parser location note) (ly:music?)
   (set! (ly:music-property note 'tweaks)
      (acons 'style 'do (ly:music-property note 'tweaks)))
   note)

TrackAVoiceAMusic = #(define-music-function (parser location inTab) (boolean?)
#{

   \clef #(if $inTab "tab" "treble_8")
   \key c \major
   \time 2/4
   \oneVoice
   <b\4 dis'\3 gis'\2 ais'\1 >8 <gis'\2 ais'\1 dis'\3 b\4 >16 <gis'\2 ais'\1 dis'\3 b\4 >16 <cis'\4 e'\3 cis''\1 a'\2 >8. <cis'\4 e'\3 cis''\1 a'\2 >16 
   <cis'\4 e'\3 a'\2 cis''\1 >16 <ais'\1 gis'\2 dis'\3 b\4 >16 <b\4 ais'\1 gis'\2 dis'\3 >8 r16 <gis'\2 b\4 ais'\1 e'\3 >8 <gis'\2 b\4 ais'\1 e'\3 >16 
   <ais'\1 gis'\2 e'\3 b\4 >8 <cis'\4 e'\3 gis'\2 cis''\1 >16 <cis'\4 e'\3 gis'\2 cis''\1 >16 <cis''\1 gis'\2 e'\3 cis'\4 >8 <gis\4 dis'\3 fis'\2 b'\1 >16 <gis\4 dis'\3 fis'\2 b'\1 >16 
   <gis\4 dis'\3 fis'\2 b'\1 >8 <cis'\4 e'\3 gis'\2 cis''\1 >16 <cis'\4 e'\3 gis'\2 cis''\1 >16 <cis'\4 e'\3 gis'\2 cis''\1 >4 
   \bar "|."
   \pageBreak
#})
TrackAVoiceBMusic = #(define-music-function (parser location inTab) (boolean?)
#{
#})

TrackAStaff = \new Staff <<
   \context Voice = "TrackAVoiceAMusic" {
      \removeWithTag #'chords
      \removeWithTag #'texts
      \TrackAVoiceAMusic ##f
   }
   \context Voice = "TrackAVoiceBMusic" {
      \removeWithTag #'chords
      \removeWithTag #'texts
      \TrackAVoiceBMusic ##f
   }
>>

 harmonies = \chordmode {
       gis4:m a4.  gis4.:m e4 cis4:m b:6 cis:m 
   }
   
TrackATabStaff = \new TabStaff \with { stringTunings = #'( 9 4 0 -5 ) } <<
   \context TabVoice = "TrackAVoiceAMusic" {
	\tabFullNotation
      \removeWithTag #'chords
      \removeWithTag #'texts
      \TrackAVoiceAMusic ##t
   }
   \context TabVoice = "TrackAVoiceBMusic" {
      \removeWithTag #'chords
      \removeWithTag #'texts
      \TrackAVoiceBMusic ##t
   }
   \new Lyrics \lyricmode {
   %\set ignoreMelismata = ##t
   If8.  you16 love4.  me,4.  won't8.   you16 let4   me   know?
   %\unset ignoreMelismata
}
>>
TrackAStaffGroup = \new StaffGroup <<
	\new ChordNames {
         \set chordChanges = ##t
         \harmonies }
   \TrackATabStaff
>>
\score {
   \TrackAStaffGroup
   \header {
      title = "" 
      composer = "" 
   }
}
````
![chorus_old](https://cloud.githubusercontent.com/assets/284394/15145322/371e1c92-16f8-11e6-9801-6cac8dce9957.jpg)

but now my code looks like this:
```
\include "../_lilypond/header"
\paper{
paper-width = 8\cm
line-width = 8\cm
paper-height=5\cm
}

\score
{  <<
   {\key c \major
   \time 2/4
   <b\4 dis'\3 gis'\2 ais'\1 >8 <gis'\2 ais'\1 dis'\3 b\4 >16 <gis'\2 ais'\1 dis'\3 b\4 >16 <cis'\4 e'\3 cis''\1 a'\2 >8. <cis'\4 e'\3 cis''\1 a'\2 >16
   <cis'\4 e'\3 a'\2 cis''\1 >16 <ais'\1 gis'\2 dis'\3 b\4 >16 <b\4 ais'\1 gis'\2 dis'\3 >8 r16 <gis'\2 b\4 ais'\1 e'\3 >8 <gis'\2 b\4 ais'\1 e'\3 >16
   <ais'\1 gis'\2 e'\3 b\4 >8 <cis'\4 e'\3 gis'\2 cis''\1 >16 <cis'\4 e'\3 gis'\2 cis''\1 >16 <cis''\1 gis'\2 e'\3 cis'\4 >8 <gis\4 dis'\3 fis'\2 b'\1 >16 <gis\4 dis'\3 fis'\2 b'\1 >16
   <gis\4 dis'\3 fis'\2 b'\1 >8 <cis'\4 e'\3 gis'\2 cis''\1 >16 <cis'\4 e'\3 gis'\2 cis''\1 >16 <cis'\4 e'\3 gis'\2 cis''\1 >4
   \bar "|."
   }
  >>
}
```

but what I get is this (which is not good at all)

![chorus_new](https://cloud.githubusercontent.com/assets/284394/15145395/a0a5c412-16f8-11e6-8a3e-51c56c0c6119.jpg)

Can you help me get back to the result I had before please?

**Comment by oliverpool** [comment]
As far as I know, we didn't touch lilypond at all.

I don't understand how your lilypond source code changed!

**Comment by casertap** [comment]
Ho sorry I changed it because it cause causing some compilation errors. let me try again and I will put tell you the errors I get.

**Comment by casertap** [comment]
I figure it out a way to update to lilypond 2.18.2
Here is the result (if someone has the same problem in the future)
````
#(define (tie::tab-clear-tied-fret-numbers grob)
   (let* ((tied-fret-nr (ly:spanner-bound grob RIGHT)))
      (ly:grob-set-property! tied-fret-nr 'transparent #t)))

\version "2.18.2"
\include "../_lilypond/header"
\paper{
paper-width = 8\cm
line-width = 8\cm
paper-height=5\cm
}
\layout {
   \context { \Score
      \override MetronomeMark #'padding = #'5
   }
   \context { \Staff
      \override TimeSignature #'style = #'numbered
      \override StringNumber #'transparent = ##t
   }
   \context { \TabStaff
      \override TimeSignature #'style = #'numbered
      \override Stem #'transparent = ##f
      \override Beam #'transparent = ##f
      \override Tie  #'after-line-breaking = #tie::tab-clear-tied-fret-numbers
   }
   \context { \StaffGroup
      \consists "Instrument_name_engraver"
   }
}
deadNote = #(define-music-function (parser location note) (ly:music?)
   (set! (ly:music-property note 'tweaks)
      (acons 'stencil ly:note-head::print
         (acons 'glyph-name "2cross"
            (acons 'style 'special
               (ly:music-property note 'tweaks)))))
   note)

palmMute = #(define-music-function (parser location note) (ly:music?)
   (set! (ly:music-property note 'tweaks)
      (acons 'style 'do (ly:music-property note 'tweaks)))
   note)

TrackAVoiceAMusic = #(define-music-function (parser location inTab) (boolean?)
#{

   \key c \major
   \time 2/4
   \oneVoice
   <b\4 dis'\3 gis'\2 ais'\1 >8 <gis'\2 ais'\1 dis'\3 b\4 >16 <gis'\2 ais'\1 dis'\3 b\4 >16 <cis'\4 e'\3 cis''\1 a'\2 >8. <cis'\4 e'\3 cis''\1 a'\2 >16
   <cis'\4 e'\3 a'\2 cis''\1 >16 <ais'\1 gis'\2 dis'\3 b\4 >16 <b\4 ais'\1 gis'\2 dis'\3 >8 r16 <gis'\2 b\4 ais'\1 e'\3 >8 <gis'\2 b\4 ais'\1 e'\3 >16
   <ais'\1 gis'\2 e'\3 b\4 >8 <cis'\4 e'\3 gis'\2 cis''\1 >16 <cis'\4 e'\3 gis'\2 cis''\1 >16 <cis''\1 gis'\2 e'\3 cis'\4 >8 <gis\4 dis'\3 fis'\2 b'\1 >16 <gis\4 dis'\3 fis'\2 b'\1 >16
   <gis\4 dis'\3 fis'\2 b'\1 >8 <cis'\4 e'\3 gis'\2 cis''\1 >16 <cis'\4 e'\3 gis'\2 cis''\1 >16 <cis'\4 e'\3 gis'\2 cis''\1 >4
   \bar "|."
   \pageBreak
#})
TrackAVoiceBMusic = #(define-music-function (parser location inTab) (boolean?)
#{
#})

TrackAStaff = \new Staff <<
   \context Voice = "TrackAVoiceAMusic" {
      \removeWithTag #'chords
      \removeWithTag #'texts
      \TrackAVoiceAMusic ##f
   }
   \context Voice = "TrackAVoiceBMusic" {
      \removeWithTag #'chords
      \removeWithTag #'texts
      \TrackAVoiceBMusic ##f
   }
>>

 harmonies = \chordmode {
       gis4:m a4.  gis4.:m e4 cis4:m b:6 cis:m
   }

TrackATabStaff = \new TabStaff <<
   \set Staff.stringTunings = \stringTuning <g c' e' a'>
   \context TabVoice = "TrackAVoiceAMusic" {
	\tabFullNotation
      \removeWithTag #'chords
      \removeWithTag #'texts
      \TrackAVoiceAMusic ##t
   }
   \context TabVoice = "TrackAVoiceBMusic" {
      \removeWithTag #'chords
      \removeWithTag #'texts
      \TrackAVoiceBMusic ##t
   }
   \new Lyrics \lyricmode {
   %\set ignoreMelismata = ##t
   If8.  you16 love4.  me,4.  won't8.   you16 let4   me   know?
   %\unset ignoreMelismata
}
>>
TrackAStaffGroup = \new StaffGroup <<
	\new ChordNames {
         \set chordChanges = ##t
         \harmonies }
   \TrackATabStaff
>>
\score {
   \TrackAStaffGroup
   \header {
      title = ""
      composer = ""
   }
}
`````

I still can not display the ukulele accords at the beginning of each song.
This is the setting I an using:
````
book:
  lang: en
  datadir: ".."
  pictures: yes
  template: data.tex
  encoding: utf8
chords:
  show: yes
  diagramreminder: all
  repeatchords: yes
  lilypond: yes
  instrument: ukulele

content:
    sort:

template:
  default.tex:
    title: "Ukulele"
authors:
  separators: ["and", "et"]
```
I tried ```  diagramreminder: important ```  with the function ```\gtab*{...}`````
Did not work.

**Comment by oliverpool** [comment]
Ok, so if I get it right, the lilypond bug was related to the lilypond software (which you solved by editing the `.ly` file): that's why I will close this issue.

The bug in the display of the ukulele is disturbing: to keep it organized, could you open a new issue?
I could be great if you could provide a minimal working example, with your `.sg` file where you have `\gtab*{...}` (actually, it should be `\utab` for ukulele)

**Comment by casertap** [comment]
```utab``` worked. Thank you for your help.
